bokehjs/src/lib/models/glyphs/marker.ts

```diff
@@ -62,20 +64,21 @@ export abstract class MarkerView extends XYGlyphView {
 
   protected _hit_point(geometry: PointGeometry): Selection {
     const {sx, sy} = geometry
+    const hit_radius = this._hit_radius == null ? this.max_size : this.max_hit_radius*2
 
-    const sx0 = sx - this.max_size
-    const sx1 = sx + this.max_size
+    const sx0 = sx - hit_radius
+    const sx1 = sx + hit_radius
     const [x0, x1] = this.renderer.xscale.r_invert(sx0, sx1)
 
-    const sy0 = sy - this.max_size
-    const sy1 = sy + this.max_size
+    const sy0 = sy - hit_radius
+    const sy1 = sy + hit_radius
     const [y0, y1] = this.renderer.yscale.r_invert(sy0, sy1)
 
     const candidates = this.index.indices({x0, x1, y0, y1})
     const indices: number[] = []
 
     for (const i of candidates) {
-      const s2 = this._size[i]/2
+      const s2 = this._hit_radius == null ? this._size[i]/2 : this._hit_radius[i]
       if (Math.abs(this.sx[i] - sx) <= s2 && Math.abs(this.sy[i] - sy) <= s2) {
```

Comment:
similar comments as above, there is too much changed here. All that needs to happen is that `s2` needs to be replaced with `s2 + this.model.hit_radius` to dilate the area that counts as a "hit"

Reply:
I note that this hit test is a bit sloppy. It's actually testing against the entire square that encloses the circle. Perpaps that was done as an optimization tradeoff, not sure.